core/domain/suggestion_services.py

```diff
@@ -1967,12 +1980,13 @@ def _get_all_question_review_stats(user_id):
         )
     )
     return [
-        _create_question_review_stats_from_models(model)
+        _create_question_review_stats_from_model(model)
         for model in question_review_stats_models
     ]
 
 
-def get_all_contributor_stats(user_id):
+def get_all_contributor_stats(
```

Comment:
Re https://github.com/oppia/oppia/pull/15917#discussion_r948652050 :

Why is fetching them all at once better than the alternative of doing so more incrementally?
Will there be any latency issues?
Is there a decision analysis for this? If not, might be worth making one.

Reply:
Fetch all at once will be needed for the badges page since we show everything at one place. I think we can use separate API calls to fetch data for the stats page based on the contribution type selected. I made a analysis here and proposed my suggestions. https://docs.google.com/document/d/10VUaO7yNMzG74a61a6NepOkUu8DOsC3gzgv8yLWaiFw/edit?usp=sharing

Reply:
Here's a thought -- will the badges page ever be the default page?

If so then yeah, two endpoints seems sensible; maybe you can cache the data in the frontend so that you don't need to refetch data you already have.

If not, then one idea might be to pre-fetch and cache the badges data while the user is on the stats page, and it's unlikely that the user will run into a lag unless they navigate to the badges page very quickly. 

What do you think?

Reply:
The default page will be the stats page where those stats tables are displayed.

And yes I agree with pre-fetching and caching badges data when the user is on the stats page.

Will it be fine if I followup this in my next PR i.e. milestone 1.6 which will basically add the controller layer related implementation?

Reply:
> The default page will be the stats page where those stats tables are displayed.

Even so, will we be providing "deep URLs" where a user can go to that URL and then automatically land on the badges page? Typically we should do this if a page has multiple tabs etc., though I suppose this is not doable immediately because even the main part of the contributor dashboard doesn't have this. But if it's something that we should do in future then maybe worth planning for it.

> Will it be fine if I followup this in my next PR i.e. milestone 1.6 which will basically add the controller layer related implementation?

That's up to @sagangwee since he's mentoring your project. I'm happy with anything you're both happy with.

Reply:
> Even so, will we be providing "deep URLs" where a user can go to that URL and then automatically land on the badges page? Typically we should do this if a page has multiple tabs etc., though I suppose this is not doable immediately because even the main part of the contributor dashboard doesn't have this. But if it's something that we should do in future then maybe worth planning for it.

In that case I will add a separate endpoint anyhow since it will be useful for future as well.

> That's up to @sagangwee since he's mentoring your project. I'm happy with anything you're both happy with. But in general if you're punting anything to later, there needs to be a TODO and an issue filed (or it needs to be tracked somewhere where it won't be missed).

Created a TODO comment and linked it [here](https://docs.google.com/document/d/16SciF9u-AngOdmvBaSyjYCAYrppR6lTE-S2vwhhh86A/edit#bookmark=id.wljrj85sobjw)

Reply:
Please create an issue and use it in the TODO, you can link the doc in the issue.

Reply:
I'm fine with doing this in M1.6.